features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/resources/ResourcesDataStoreManager.kt

```diff
@@ -0,0 +1,164 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.sessionreplay.internal.resources
+
+import android.text.format.DateUtils
+import com.datadog.android.api.feature.FeatureScope
+import com.datadog.android.api.feature.FeatureSdkCore
+import com.datadog.android.api.storage.datastore.DataStoreCallback
+import com.datadog.android.core.internal.persistence.Deserializer
+import com.datadog.android.core.persistence.Serializer
+import com.datadog.android.core.persistence.datastore.DataStoreContent
+import com.datadog.android.sessionreplay.internal.ResourcesFeature.Companion.SESSION_REPLAY_RESOURCES_FEATURE_NAME
+import java.util.Collections
+
+@Suppress("TooManyFunctions")
+internal class ResourcesDataStoreManager(
+    private val featureSdkCore: FeatureSdkCore,
+    private val resourceHashesSerializer: Serializer<Set<String>>,
+    private val resourcesHashesDeserializer: Deserializer<String, Set<String>>,
+    private val updateDateSerializer: Serializer<Long>,
+    private val updateDateDeserializer: Deserializer<String, Long>,
+    private val featureScope: FeatureScope? = featureSdkCore.getFeature(
+        SESSION_REPLAY_RESOURCES_FEATURE_NAME
+    )
+) {
+    private val knownResources = Collections.synchronizedSet(mutableSetOf<String>())
+
+    init {
+        initializeManager()
+    }
+
+    internal fun wasResourcePreviouslySent(resourceHash: String): Boolean =
+        knownResources.contains(resourceHash)
+
+    internal fun store(resourceHash: String) {
+        knownResources.add(resourceHash)
+        writeResources()
+    }
+
+    // region internal
+
+    private fun initializeManager() {
+        fetchLastUpdateDate(
+            onFetchSuccessful = { dataStoreContentUpdateDate ->
+                onFetchLastUpdateDateSuccess(dataStoreContentUpdateDate?.data)
+            },
+            onFetchFailure = {
+                createOrOverwriteLastUpdateDateFile(featureScope)
+            }
+        )
+    }
+
+    private fun writeResources() {
+        featureScope?.dataStore?.setValue(
+            data = knownResources,
+            key = DATASTORE_HASHES_CONTENT_FILENAME,
+            serializer = resourceHashesSerializer
+        )
+    }
+
+    private fun createOrOverwriteLastUpdateDateFile(featureScope: FeatureScope?) {
```

Comment:
> [!TIP]
> the name of this method seems a bit verbose, but more importantly makes an assumption that the feature data store is stored in a file.

Reply:
👍  you're right. I'll change to more generic names that don't assume file